src/herder/test/HerderTests.cpp

```diff
@@ -1318,49 +1301,54 @@ testSCPDriver(uint32 protocolVersion, uint32_t maxTxSize, size_t expectedOps,
 
         SECTION("valid")
         {
-            auto nomV = makeTxPair(herder, txSet0, ct, true);
+            auto nomV = makeTxPair(herder, txSet0, ct);
             REQUIRE(scp.validateValue(seq, nomV.first, true) ==
                     SCPDriver::kFullyValidatedValue);
 
-            auto balV = makeTxPair(herder, txSet0, ct, signedBallots);
+            auto balV = makeTxPair(herder, txSet0, ct);
             REQUIRE(scp.validateValue(seq, balV.first, false) ==
                     SCPDriver::kFullyValidatedValue);
         }
         SECTION("invalid")
         {
-            auto nomV = makeTxPair(herder, txSet0, ct, false);
-            REQUIRE(scp.validateValue(seq, nomV.first, true) ==
-                    SCPDriver::kInvalidValue);
-
-            auto balV = makeTxPair(herder, txSet0, ct, !signedBallots);
-            REQUIRE(scp.validateValue(seq, balV.first, false) ==
-                    SCPDriver::kInvalidValue);
-
-            auto p = makeTxPair(herder, txSet0, ct, true);
-            StellarValue sv;
-            xdr::xdr_from_opaque(p.first, sv);
 
-            auto checkInvalid = [&](StellarValue const& sv) {
+            auto checkInvalid = [&](StellarValue const& sv,
```

Comment:
Actually why not wrap this whole thing and test both variations (I don't see a reason to only check for bad signatures etc only for the ballot protocol case)

```
            SECTION("Nomination")
            {
                TestInvalid(true);
            }
            SECTION("Ballot")
            {
                TestInvalid(false);
            }
```

Reply:
done!